src/modules/booking/components/service-booking-modal/service-booking-modal.tsx

```diff
@@ -0,0 +1,133 @@
+import { BaseCardWithRadioButton } from '../booking-card-radio-button';
+// type
+import { type ModalProps } from '@/modules/core/components/dialog/dialog.interface';
+// utils
+import { trpc } from '@/modules/core/utils/trpc.utils';
+
+import styles from './service-booking-modal.module.scss';
+
+export const ServiceBookingModal: FC<Omit<ModalProps, 'children'>> = (
+  props
+) => {
+  // state
+  const [value, setValue] = useState<string>('');
+
+  const [step, setStep] = useState<'service' | 'datetime' | 'confirmation'>(
+    'service'
+  );
+
+  const handleNext = () => {
+    if (step === 'service') {
+      setStep('datetime');
+    } else if (step === 'datetime') {
+      setStep('confirmation');
+    }
+  };
+
+  const handleBack = () => {
+    if (step === 'confirmation') {
+      setStep('datetime');
+    } else if (step === 'datetime') {
+      setStep('service');
+    }
+  };
+
+  const { data: serviceList } = trpc.serviceOnProfessional.list.useQuery({
+    limit: 10,
+    offset: 0,
+  });
+
+  return (
+    <Dialog {...props}>
+      <div className={styles.container}>
+        <div className={styles.header}>
+          <Typography variant='subtitle'>New booking</Typography>
+          <Button
+            onClick={() => props.onOpenChange && props.onOpenChange(false)}
+            variant='unstyled'
+            icon='close'
+          />
+        </div>
+        <div className={styles.labelContainer}>
+          <Avatar />
+
+          <div className={styles.lableDescr}>
+            <Typography variant='body1'>Tennisha’s Beauty</Typography>
+            <Typography className={styles.lableAddress} variant='small'>
+              Greyhound Dr, Bradford BD7 1NQ
+            </Typography>
+          </div>
+        </div>
+        <div className={styles.stepper}>
+          <Stepper
+            steps={[
+              {
+                text: 'Service',
+                value: 'service',
+              },
+              {
+                text: 'Date & Time',
+                value: 'datetime',
+              },
+              {
+                text: 'Confirmation',
+                value: 'confirmation',
+              },
+            ]}
+            value={step}
+          />
+        </div>
+
+        <div className={styles.contentContainer}>
+          {step === 'service' && (
+            <RadioButton.Group value={value} onChange={setValue} name='cards'>
+              <div className={styles.baseCardContainer}>
+                {serviceList?.map((service) => (
+                  <BaseCardWithRadioButton
+                    key={service.id}
+                    value={value}
+                    serviceOnProfessional={service}
+                    onClick={(currentValue) => setValue(currentValue)}
+                  />
+                ))}
+              </div>
+            </RadioButton.Group>
+          )}
+          {step === 'confirmation' && <BookingForm />}
+          {step === 'datetime' && <BookingCalendar />}
```

Comment:
Let's use approach that we discussed in telegram. 

Define such object:

```tsx
type BookingStep = 'service-select' | 'confirmation' | 'datetime';

const stepsData: Record<BookingStep, { Step: FC }> = {
  service: { Step: ServiceOnProfessionalSelect }, // this is the new component that you need to create and put 93-104 lines there.
  confirmation: { Step: BookingForm },
  datetime: { Step: BookingCalendar }, // don't forget rename it to DaySelect  
}
```

then, simply extract Step component:

```
export const ServiceBookingModal ... => {
  ....

  const { Step } = stepsData[step];
  
  return (
    <Dialog ...>
      ....
      <div className={styles.contentContainer}>
        <Step />
      </div>
      ....
    </Dialog>
  )
}
```

After this everything should work the same way as it is.